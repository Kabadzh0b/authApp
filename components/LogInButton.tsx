import React from 'react';
import {useActions} from "../hooks/useActions";
import {Pressable, StyleSheet, Text} from "react-native";


interface ButtonProps {
    email: string,
    password: string,
    width: number,
    setValidEmail: (value: boolean) => void,
    setValidPassword: (value: boolean) => void,
}

function LogInButton({email, password, width, setValidEmail, setValidPassword}: ButtonProps) {
    const {login} = useActions();

    return (
        <Pressable style={[styles.buttonStyle, {width: width}]} onPress={(e) => {
            e.preventDefault();
            let isCorrect = true;
            if (email === "") {
                setValidEmail(false);
                isCorrect = false;
            }
            if (password === "") {
                setValidPassword(false);
                isCorrect = false;
            }
            if (isCorrect) login(email, password);
        }}>
            <Text style={styles.titleStyle}>Увійти</Text>
        </Pressable>
    );
}

const styles = StyleSheet.create({
    buttonStyle: {
        backgroundColor: 'rgba(111, 202, 186, 1)',
        borderRadius: 5,
        height: 50,
        marginVertical: 10,
        maxWidth: 500,
        justifyContent: "center",
        alignItems: "center",
    },
    titleStyle: {
        fontWeight: 'bold',
        fontSize: 23,
    }
})

export default LogInButton;
