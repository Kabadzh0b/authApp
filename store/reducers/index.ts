import {combineReducers} from "redux";
import {authReducer} from "./authReduces";


export const rootReducer = combineReducers({
    auth: authReducer,
});

export type RootState = ReturnType<typeof rootReducer>;
