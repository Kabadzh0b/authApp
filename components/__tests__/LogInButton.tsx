import React from "react";
import renderer from "react-test-renderer";
import LogInButton from "../LogInButton";
import {Provider} from "react-redux";
import {store} from "../../store";
it('default renders correctly', () => {
    const button = renderer
        .create(
            <Provider store={store}>
                <LogInButton email={"dsd"} password={"dsd"} width={200} setValidEmail={()=>{}}
                             setValidPassword={()=>{}}/>
            </Provider>
        )
        .toJSON();
    expect(button).toMatchSnapshot();
})

it('empty renders correctly', () => {
    const button = renderer
        .create(
            <Provider store={store}>
                <LogInButton email={""} password={""} width={200} setValidEmail={()=>{}}
                             setValidPassword={()=>{}}/>
            </Provider>
        )
        .toJSON();
    expect(button).toMatchSnapshot();
})

it('big width renders correctly', () => {
    const button = renderer
        .create(
            <Provider store={store}>
                <LogInButton email={""} password={""} width={1000} setValidEmail={()=>{}}
                             setValidPassword={()=>{}}/>
            </Provider>
        )
        .toJSON();
    expect(button).toMatchSnapshot();
})
