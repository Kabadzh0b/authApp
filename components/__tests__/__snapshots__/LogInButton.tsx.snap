// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`big width renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "rgba(111, 202, 186, 1)",
        "borderRadius": 5,
        "height": 50,
        "justifyContent": "center",
        "marginVertical": 10,
        "maxWidth": 500,
      },
      {
        "width": 1000,
      },
    ]
  }
>
  <Text
    style={
      {
        "fontSize": 23,
        "fontWeight": "bold",
      }
    }
  >
    Увійти
  </Text>
</View>
`;

exports[`default renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "rgba(111, 202, 186, 1)",
        "borderRadius": 5,
        "height": 50,
        "justifyContent": "center",
        "marginVertical": 10,
        "maxWidth": 500,
      },
      {
        "width": 200,
      },
    ]
  }
>
  <Text
    style={
      {
        "fontSize": 23,
        "fontWeight": "bold",
      }
    }
  >
    Увійти
  </Text>
</View>
`;

exports[`empty renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "rgba(111, 202, 186, 1)",
        "borderRadius": 5,
        "height": 50,
        "justifyContent": "center",
        "marginVertical": 10,
        "maxWidth": 500,
      },
      {
        "width": 200,
      },
    ]
  }
>
  <Text
    style={
      {
        "fontSize": 23,
        "fontWeight": "bold",
      }
    }
  >
    Увійти
  </Text>
</View>
`;

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "rgba(111, 202, 186, 1)",
        "borderRadius": 5,
        "height": 50,
        "marginHorizontal": 50,
        "marginVertical": 10,
        "maxWidth": 500,
      },
      {
        "width": 200,
      },
    ]
  }
>
  <Text
    style={
      {
        "fontSize": 23,
        "fontWeight": "bold",
      }
    }
  >
    Увійти
  </Text>
</View>
`;
